### ISSUE TITLE
Can't override space-before-function-paren from Google config

### ISSUE THREAD

**Issue by dandv** [eslint/eslint#7255]
ESLint 3.6.1, Node 6.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```js
module.exports = {
  "extends": "google",
  "space-before-function-paren": [
    "error",
    {
      "anonymous": "always",
      "named": "never"
    }
  ],
};
```

**What did you do? Please include the actual source code causing the issue.**
```js
let x = function (y) {
  console.log('foo');
};
console.log(x);
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
  1:17  error  Unexpected space before function parentheses  space-before-function-paren

✖ 1 problem (1 error, 0 warnings)
```

I've flipped the values in .eslintrc to `{"anonymous": "never", "named": "always"}` but get the same error.

**Comment by vitorbal** [comment]
@dandv thanks for reaching out! The rules you want to override need to be inside a `"rules"` property in your config file. That's probably why the rule is not being overridden. Try this:

```javascript
module.exports = {
  "extends": "google",
  "rules": {
      "space-before-function-paren": [
        "error",
        {
          "anonymous": "always",
          "named": "never"
        }
      ]
  }
};
```

**Comment by dandv** [comment]
That worked, thanks!
